storage.go

```diff
@@ -559,10 +592,30 @@ func (s *Storage) stat(ctx context.Context, path string, opt pairStorageStat) (o
 	o = s.newObject(true)
 	o.ID = rp
 	o.Path = path
-	if opt.HasObjectMode && opt.ObjectMode.IsDir() {
-		o.Mode |= ModeDir
-	} else {
-		o.Mode |= ModeRead
+
+	if output.XQSMetaData != nil {
+		metadata := *output.XQSMetaData
+		// By calling `HeadObject`, the first letter of the `key` of the object metadata will be capitalized.
+		if v, ok := metadata["X-Qs-Meta-Bs-Link-Target"]; ok {
```

Comment:
How about uniform metadata naming here and also in `createLink`. And we can extract it as a const value.

Reply:
If we want to set it as a constant, we need to set two constants. One is the name of the metadata when it is set, and the other is the name of the metadata obtained through `HeadObject`. Because it must be set with the lowercase prefix `x-qs-meta`, the metadata name obtained from `HeadObject` will have all initials in upper case.

Reply:
According to [API doc](https://docs.qingcloud.com/qingstor/api/common/metadata), since HTTP headers field name is case-insensitive, the key of custom metadata is also case-insensitive.

Reply:
I've tried all cases, and when the prefix is not lowercase `x-qs-meta` it gives a disallowed error. I think what it means by case-insensitive is that the prefix is followed by a custom name. I've also tried this, and it doesn't matter if the custom name after the prefix is uppercase or lowercase, it ends up with only the first letter in upper case.

Reply:
Perhaps we could extract them as constant values like this.
```go
 const (
	linkMetaAtSet = "x-qs-meta-bs-link-target"
	linkMetaAtGet = "X-Qs-Meta-Bs-Link-Target"
)
```

Reply:
> Perhaps we could extract them as constant values like this.
> 
> ```go
>  const (
> 	linkMetaAtSet = "x-qs-meta-bs-link-target"
> 	linkMetaAtGet = "X-Qs-Meta-Bs-Link-Target"
> )
> ```

negative, so wired.

I think it's a problem introduced by qingstor go sdk: https://github.com/qingstor/qingstor-sdk-go/blob/52e18055f9371d57663cab22381b1928bf392b9b/utils/check_metadata.go#L21-L23

```go
var metadataPrefix = "x-qs-meta-"

// validKeyChars contains all valid key char: (a-z, A-Z, -, _, .)
var validKeyChars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-_."

// IsMetaDataValid check whether the metadata-KV follows rule in API document
// https://docs.qingcloud.com/qingstor/api/common/metadata.html#%E5%AF%B9%E8%87%AA%E5%AE%9A%E4%B9%89%E5%85%83%E6%95%B0%E6%8D%AE%E7%9A%84%E9%99%90%E5%88%B6
func IsMetaDataValid(XQSMetaData *map[string]string) error {
	metadataSize := 0

	for k, v := range *XQSMetaData {
		if !strings.HasPrefix(k, metadataPrefix) {
			return newMetaDataInvalidError(k, v)
		}
```

In which they hardcoded `x-qs-meta`.

Reply:
I open an issue for qingstor-sdk-go https://github.com/qingstor/qingstor-sdk-go/issues/132

Reply:
The issue has been resolved in https://github.com/qingstor/qingstor-sdk-go/commit/5afc7bc0d697935f08cbcd755493fa3c5298ffc8